src/main/java/adris/altoclef/tasks/misc/GetToOuterEndIslandsTask.java

```diff
@@ -0,0 +1,120 @@
+package adris.altoclef.tasks.misc;
+
+import adris.altoclef.AltoClef;
+import adris.altoclef.Debug;
+import adris.altoclef.tasks.InteractWithBlockTask;
+import adris.altoclef.tasks.movement.GetToYTask;
+import adris.altoclef.tasks.resources.GetBuildingMaterialsTask;
+import adris.altoclef.tasks.speedrun.BeatMinecraft2Task;
+import adris.altoclef.tasks.squashed.CataloguedResourceTask;
+import adris.altoclef.tasksystem.Task;
+import adris.altoclef.util.Dimension;
+import adris.altoclef.util.ItemTarget;
+import adris.altoclef.util.baritone.GoalAnd;
+import adris.altoclef.util.helpers.StorageHelper;
+import adris.altoclef.util.helpers.WorldHelper;
+
+import baritone.api.pathing.goals.GoalComposite;
+import baritone.api.pathing.goals.GoalGetToBlock;
+import baritone.api.pathing.goals.GoalYLevel;
+import net.minecraft.block.Blocks;
+import net.minecraft.item.Items;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.Vec3d;
+import net.minecraft.util.math.Vec3i;
+
+public class GetToOuterEndIslandsTask extends Task {
+    private Task _beatTheGame;
+    private boolean _pathingToGateway = false;
+    public final Vec3i[] OFFSETS = {
+            new Vec3i(1, -1, 1),
+            new Vec3i(1, -1, -1),
+            new Vec3i(-1, -1, 1),
+            new Vec3i(-1, -1, -1),
+            new Vec3i(2, -1, 0),
+            new Vec3i(0, -1, 2),
+            new Vec3i(-2, -1, 0),
+            new Vec3i(0, -1, -2)
+    };
+    public GetToOuterEndIslandsTask() {
+
+    }
+
+    @Override
+    protected void onStart(AltoClef mod) {
+        mod.getBehaviour().push();
+        mod.getBlockTracker().trackBlock(Blocks.END_GATEWAY);
+        _beatTheGame = new BeatMinecraft2Task();
+    }
+
+    @Override
+    protected Task onTick(AltoClef mod) {
+        if (isFinished(mod)) {
+            setDebugState("What?");
+            return null;
+        }
+        if (mod.getBlockTracker().anyFound(Blocks.END_GATEWAY)) {
+            if (!mod.getItemStorage().hasItemInventoryOnly(Items.ENDER_PEARL)) {
+                setDebugState("Getting an ender pearl");
+                return new CataloguedResourceTask(new ItemTarget(Items.ENDER_PEARL, 1));
```

Comment:
This may already work just fine, so take this comment as a suggestion rather than a change requisition. 

you may want to get the enderman hashmap from mob defence so that you can always find one. Could be wrong here.

Also, you should get more than 1 pearl. You don't know how many islands you will need to hop.

Reply:
Not thinking longterm yet, just want to get to the outer end island

Reply:
Don't enderman spawn everywhere in the outer islands? If you need pearls just find one there

Reply:
Exactly